sdk/src/main/java/com/podio/sdk/androidasynchttp/AndroidAsyncHttpRequest.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2015 Citrix Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.podio.sdk.androidasynchttp;
+
+import android.content.Context;
+
+import com.google.gson.JsonSyntaxException;
+import com.loopj.android.http.AsyncHttpClient;
+import com.loopj.android.http.BaseJsonHttpResponseHandler;
+import com.loopj.android.http.RequestParams;
+import com.podio.sdk.ApiError;
+import com.podio.sdk.ConnectionError;
+import com.podio.sdk.NoResponseError;
+import com.podio.sdk.Podio;
+import com.podio.sdk.PodioError;
+import com.podio.sdk.Request;
+import com.podio.sdk.Session;
+import com.podio.sdk.internal.CallbackManager;
+import com.podio.sdk.json.JsonParser;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.SocketTimeoutException;
+import java.util.Calendar;
+import java.util.concurrent.TimeUnit;
+
+import cz.msebera.android.httpclient.Header;
+import cz.msebera.android.httpclient.conn.ConnectTimeoutException;
+import cz.msebera.android.httpclient.message.BasicHeader;
+
+/**
+ * @author Tobias Lindberg
+ */
+public class AndroidAsyncHttpRequest<T> implements Request<T>, Request.SessionListener {
+    private long TEN_MINUTES = 600000;
+    private static final String CONTENT_TYPE = "application/json; charset=UTF-8";
+
+    private AsyncHttpClient client;
+    private Context context;
+    private String url;
+    private File file;
+    private final Class<T> classOfResult;
+    private CallbackManager<T> callbackManager;
+
+    private T result;
+    private boolean isDone;
+    private PodioError error;
+
+    @Override
+    public T waitForResult(long maxSeconds) throws PodioError {
+        // This is still awkward as we might end up blocking the delivery of
+        // a client side error (see
+        try {
+            wait(TimeUnit.SECONDS.toMillis(Math.max(maxSeconds, 0)));
+        } catch (InterruptedException e) {
+            callbackManager.deliverError(e);
+        }
+
+        if (error != null) {
+            throw error;
+        }
+
+        return result;
+    }
+
+    @Override
+    public Request<T> withResultListener(ResultListener<T> contentListener) {
+        callbackManager.addResultListener(contentListener, isDone, result);
```

Comment:
The listener on line 80 is results only and then errors are returned with the error listener. This is done same way as it was done in volley request.